app/js/lib/foxbox/settings.js

```diff
@@ -203,4 +187,33 @@ export default class Settings extends EventDispatcher {
   get apiVersion() {
     return API_VERSION;
   }
+
+  clear() {
+    return new Promise((resolve) => {
+      this.session = null;
+      // @todo Remove only the items set here.
+      storage.clear();
+      resolve();
+    });
+  }
+
+  [p.updateSetting](key, value) {
+    if (this[p[key]] === value) {
+      return;
+    }
+
+    this[p[key]] = value;
+
+    if (value !== null && value !== undefined) {
+      storage.setItem(`${PREFIX}${key}`, value);
+    } else {
+      storage.removeItem(`${PREFIX}${key}`);
+    }
+
+    this.emit('setting', { key, value });
+    this.emit(
```

Comment:
Yeah, why not, but it's a good candidate for a separate PR/issue as it will involve a bit of additional code (to parse name of the key to remove `PREFIX` part + to deal correctly with boolean strings). Could you please issue and I'll look into it once my wips are landed?